datafusion/optimizer/src/optimizer.rs

```diff
@@ -144,6 +144,10 @@ impl Optimizer {
             Arc::new(DecorrelateWhereIn::new()),
             Arc::new(ScalarSubqueryToJoin::new()),
             Arc::new(SubqueryFilterToJoin::new()),
+            // simplify expressions does not simplify expressions in subqueries, so we
```

Comment:
I think it's a solution to resolve this.
But We also has a way to resolve this issue by supporting the `subquery` in the rewriter of `simplify expression`.
Why not support the `subquery` in the rewriter of simplify expression?

Reply:
This PR just reverts the regression and adds a regression test, but I agree, would be better to handle it in `simplify_expressions`. I filed https://github.com/apache/arrow-datafusion/issues/3770 for this.

Reply:
I agree #3770 is a better long term solution